modules/ROOT/pages/dataweave-formats-java.adoc

```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
 | MIME Type
 |`*/java`
 |===
+
+
+[[java_support]]
+== Java Support
+
+Starting with version 2.6, DataWeave supports Java 17. Java 17 adds restrictions in encapsulation and reflective access that affects the xref:dataweave-formats-java.adoc[]. Because DataWeave uses Java's reflection API to read and write Java objects, ensure that the objects your application uses are Plain Old Java Objects (POJOs) containing:
+
+* Default constructor
+* Getters for all properties
+* Setters for all properties
+
+[[java17_examples]]
+== Examples of Java Support
+
+The following examples run using Java 17 and using xref:dataweave-versioning-behavior.adoc[language level 2.6].
+
+* <<example4>>
+* <<example5>>
+
+[[example4]]
+=== Example: Create a Java Object Instance
+
+This example shows how to create a Java object instance.
+
+==== User class
+
+The Java example uses the following class:
+
+[source,java,linenums]
+----
+public class User {
+   private String name;
+   private String lastName;
+   public User(){
+   }
+   public String getName(){
+       return name;
+   }
+   public String getLastName(){
+       return lastName;
+   }
+}
+----
+
+==== DataWeave Script
+
+[source,java,linenums]
+----
+{name: "Tomo", lastName: "Chibana" } as Object {class: "User"}
+----
+
+==== Output
+
+The expected output is an instance of the user class. Because the user class definition doesn't have the proper setters, the output is `IllegalAccessException`.   
+
+[source,java,linenums]
+----
+java.lang.IllegalAccessException - class 
+org.mule.weave.v2.module.pojo.reader.PropertyDefinition (in module org.mule.weave.module.java) cannot access a member of class User with modifiers "private
+----
+
+[[example5]]
+=== Example: Access the Values of Java Properties
+
+This example shows how to access the values of Java properties.
+
+==== Item class
+
+The Java example uses the following class:
+
+[source,java,linenums]
+----
+public class Item {
+	private String name;
+	private int quantity;
+	
+	public Item() {
+	}
+    public Item(String name, int quantity) {
+		this.name = name;
+		this.quantity = quantity;
+	}
+	public void setName(String name) {
+		this.name = name;
+	}
+	public void setQuantity(int quantity) {
+		this.quantity = quantity;
+	}
+}
+----
+
+==== Input
+
+----
+new Item("Pencil", 5)
+----
+
+==== DataWeave Script
+
+[source,dataweave,linenums]
+----
+output application/json
+---
+{ 
+    "name": "payload.name",
+    "quantity": "payload.quantity"
+}
+----
+
+==== Output
+
+The expected output is a JSON.  
```

Comment:
```suggestion
The expected output is JSON.  
```

Reply:
Not sure if that's correct. Should it be "in JSON format" (or something like that)?